#!/usr/bin/env python
PACKAGE = "crazyflie_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Wdiag_xq", double_t, 5.0, "Weight of xq", 0.0, 0.1, 1000)
gen.add("Wdiag_yq", double_t, 5.0, "Weight of yq", 0.0, 0.1, 1000)
gen.add("Wdiag_zq", double_t, 10.0, "Weight of zq", 0.0, 0.1, 1000)
gen.add("Wdiag_q1", double_t, 1e-1, "Weight of q1", 0.0, 0.1, 1000)
gen.add("Wdiag_q2", double_t, 1e-1, "Weight of q2", 0.0, 0.1, 1000)
gen.add("Wdiag_q3", double_t, 1e-1, "Weight of q3", 0.0, 0.1, 1000)
gen.add("Wdiag_q4", double_t, 1e-1, "Weight of q4", 0.0, 0.1, 1000)
gen.add("Wdiag_vbx", double_t, 1.0, "Weight of vbx", 0.0, 0.1, 1000)
gen.add("Wdiag_vby", double_t, 1.0, "Weight of vby", 0.0, 0.1, 1000)
gen.add("Wdiag_vbz", double_t, 2.0, "Weight of vbz", 0.0, 0.1, 1000)
gen.add("Wdiag_wx", double_t, 1.5, "Weight of wx", 0.0, 0.1, 1000)
gen.add("Wdiag_wy", double_t, 1.5, "Weight of wy", 0.0, 0.1, 1000)
gen.add("Wdiag_wz", double_t, 5.0, "Weight of wz", 0.0, 0.1, 1000)
gen.add("Wdiag_w1", double_t, 2.0, "Weight of w1", 0.0, 0.1, 1000)
gen.add("Wdiag_w2", double_t, 2.0, "Weight of w2", 0.0, 0.1, 1000)
gen.add("Wdiag_w3", double_t, 2.0, "Weight of w3", 0.0, 0.1, 1000)
gen.add("Wdiag_w4", double_t, 2.0, "Weight of w4", 0.0, 0.1, 1000)


exit(gen.generate(PACKAGE, "crazyflie_controller", "WeightMatrices"))
